### ISSUE TITLE
Type alias named `undefined` is allowed, but doesn't work

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#3087]
```ts
type undefined = string;
var x: undefined; // Error, cannot find name 'undefined'
```
Expected: Type alias should be an error, or the type annotation should work
